### PR TITLE
Add WhiteSpaceAroundComma fixer

### PR DESCRIPTION
**Pull request by SpacePossum**
Early opening so I've travis reports.

I really don't like the comma-in-array fixer making multi line arrays into single line ones.
This fixer fixes that and a bunch more cases.

Some behavior changes:

New vs. old
```php
                '<?php function xyz4(
                    $a=10,  //comment1
                    $b=2000, //comment2 // now adds a space here
                    $c=30) {
                }',
                '<?php function xyz4(
                    $a=10,  //comment1
                    $b=2000,//comment2
                    $c=30) {
                }',
```

```php
        // from MethodArgumentSpaceFixerTest 
        '<?php list($path, $mode, ) = foo();', // no removes the space after the last comma
        '<?php list($path, $mode,) = foo();',
```

Fixes to:

```php
list($a, , , , $b) = a();
``` 

(some for arrays etc.)

### CODE REVIEW

**Comment by SpacePossum** [comment]
any opinions on the second the todo item?

**Comment by gharlan** [comment]
> `list($a, , , , $b) = a();`

👍

**Comment by keradus** [comment]
What is the state of this PR ?

**Comment by SpacePossum** [comment]
>What is the state of this PR ?

ready for review
(but we could also focus on bugs for 1.x and 2.x release before picking up new features.)

**Comment by keradus** [comment]
Yes and no, as this feature is BC breaker (as it removes some fixers)

**Comment by keradus** [comment]
1. Please take a look:

------------------

2. Please rebase

-------------------

3.
This code:
```php
list(/$a, $b/**/,/**/$c, $d,) = foo();
```

is changed into:
```php
list(/$a, $b, /**//**/$c, $d,) = foo();
```

So comment has been assigned into different item.

Also this:
```php
                    function foo(
                        /**/ $a
                        $b /**/,
                        $c, /**/
                        $d
                    ) {}

File: tests/FixerFactoryTest.php [hunk 1]
```diff
@@ -268,6 +268,12 @@ public function getFixersPriorityCases()
             array($fixers['no_empty_statement'], $fixers['switch_case_semicolon_to_colon']),  // tested also in: no_empty_statement,switch_case_semicolon_to_colon.test
             array($fixers['no_empty_statement'], $fixers['no_trailing_whitespace']), // tested also in: no_empty_statement,no_trailing_whitespace.test
             array($fixers['no_empty_statement'], $fixers['no_whitespace_in_blank_lines']), // tested also in: no_empty_statement,no_whitespace_in_blank_lines.test
+            array($fixers['no_trailing_comma_in_list_call'], $fixers['white_space_around_comma']), // no hard dependency; for performance reasons only
```
Comment (keradus) [inline tests/FixerFactoryTest.php]: I love this comment !

**Comment by SpacePossum** [comment]
@keradus Thanks for the help!
A - done
B - done
C - I like the changes made by the fixer, but I see your concern about the aligned arrays. 

I reconsider this fixer, more opinions are welcome!

(I'll focus development time on the PHP7 import bug fix PR first for now, but feel free to discuss this further!)

**Comment by SpacePossum** [comment]
fabbot is wrong, seems like PSR0 v.s. 4 issue

**Comment by keradus** [comment]
> (I'll focus development time on the PHP7 import bug fix PR first for now, but feel free to discuss this further!)

That's exactly why I point out this fixer to have it ready and merged

> I  like the changes made by the fixer, but I see your concern about the aligned arrays.

About arrays: if the change won't bother anyone I will merge this PR as is.
But what about comments movement ?


> fabbot is wrong

it is indeed!

**Comment by SpacePossum** [comment]
Im closing this after some testing done by @keradus and myself.
I would like to have a fixer doing this; https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1860/files#diff-76aed87b6b6ca249ca7e21a6ffa6fbdaR102
however this PR has grown beyond that.
Thanks all for testing and reviewing!

**Comment by keradus** [comment]
As discussed, at least for now ;)

**Comment by keradus** [comment]
after discussion with @SpacePossum closing this one again.

**Pull request closed without merging**